btcec/signature_test.go

```diff
@@ -491,3 +493,88 @@ func TestSignCompact(t *testing.T) {
 		testSignCompact(t, name, btcec.S256(), data, compressed)
 	}
 }
+
+func verifyRFC6979Nonce(hexKey string, msg string, hexNonce string, t *testing.T) {
+	pkBytes, _ := hex.DecodeString(hexKey)
+	nonce, _ := hex.DecodeString(hexNonce)
+	hasher := fastsha256.New()
+	hasher.Write([]byte(msg))
+	hash := hasher.Sum(nil)
+	actualNonceBigInt := btcec.NonceRFC6979(pkBytes, hash)
+	actualNonce := actualNonceBigInt.Bytes()
+	if !bytes.Equal(actualNonce, nonce) {
+		t.Errorf("Nonce for message '%s' is incorrect: %s (expected %s)", msg, hex.EncodeToString(actualNonce), hex.EncodeToString(nonce))
+	}
+}
+
+func verifyRFC6979Signature(hexKey string, msg string, hexSignature string, t *testing.T) {
+	pkBytes, _ := hex.DecodeString(hexKey)
+	privKey, _ := btcec.PrivKeyFromBytes(btcec.S256(), pkBytes)
+	sigBytes, _ := hex.DecodeString(hexSignature)
+	hasher := fastsha256.New()
+	hasher.Write([]byte(msg))
+	hash := hasher.Sum(nil)
+	actualSig, _ := privKey.Sign(hash)
+	actualSigBytes := actualSig.Serialize()
+	if !bytes.Equal(actualSigBytes, sigBytes) {
+		t.Errorf("Signature for message '%s' is incorrect: %s (expected %s)", msg, hex.EncodeToString(actualSigBytes), hex.EncodeToString(sigBytes))
+	}
+}
+
+func TestRFC6979(t *testing.T) {
+	// Test vectors matching Trezor and CoreBitcoin implementations.
+	// - https://github.com/trezor/trezor-crypto/blob/9fea8f8ab377dc514e40c6fd1f7c89a74c1d8dc6/tests.c#L432-L453
+	// - https://github.com/oleganza/CoreBitcoin/blob/e93dd71207861b5bf044415db5fa72405e7d8fbc/CoreBitcoin/BTCKey%2BTests.m#L23-L49
+	verifyRFC6979Nonce(
+		"cca9fbcc1b41e5a95d369eaa6ddcff73b61a4efaa279cfc6567e8daa39cbaf50",
```

Comment:
Because these are hard coded strings in the source code, I'd prefer all of these use a function like the following `decodeHex` which panics if any errors occur rather than ignoring them.

```Go
// decodeHex converts the passed hex string into a big integer pointer and will
// panic is there is an error.  This is only provided for the hard-coded
// constants so errors in the source code can bet detected. It will only (and
// must only) be called for initialization purposes.
func decodeHex(hexStr string) []byte {
	hexBytes, err := hex.DecodeString(hexStr)
	if err != nil {
		panic("invalid hex in source file: " + hexStr)
	}
	return hexBytes
}
```